### PR TITLE
Remove federated url path if set

### PR DESCRIPTION
**Pull request by ohadbitt**
#### Pull Request Description

**Features:**
- Strip Fed=true from URL

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 1]
```diff
@@ -42,6 +42,10 @@ public String getClusterUrl() {
         return clusterUri;
     }
 
+    void setClusterUrl(String clusterUri) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java:45]: This isn't relevant to your change, but I just wanted to say that I suggest we make public all methods here that are part of its API (which doesn't include setClusterUrl). I think we want the other methods to be accessible even to other packages (even though Java SDK isn't currently organized).

File: data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java [hunk 2]
```diff
@@ -158,4 +159,13 @@ void createWithAadTokenProviderAuthentication(){
         Assertions.assertDoesNotThrow( () -> ConnectionStringBuilder
                 .createWithAadTokenProviderAuthentication("resource.uri", () -> "token"));
     }
+
+    @Test
+    @DisplayName("validate ClientImpl strips fed=true")
+    void stripFedereatedAuthFromCSB() throws URISyntaxException {
```
Comment (yihezkel) [inline data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java:165]: Since we're trying to get all our code more into testing, I suggest also adding a negative test for this field.

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 3]
```diff
@@ -33,7 +35,16 @@
     private final String applicationNameForTracing;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
-        clusterUrl = csb.getClusterUrl();
+        String url = csb.getClusterUrl();
+        URI clusterUri = new URI(url);
+        String host = clusterUri.getHost();
+        String auth = clusterUri.getAuthority().toLowerCase();
+        if (host == null && auth.endsWith(";fed=true")) {
+            url = new URIBuilder().setScheme(clusterUri.getScheme()).setHost(auth.substring(0, clusterUri.getAuthority().indexOf(";fed=true"))).toString();
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java:43]: Need null checks, though I'd personally instead add validation to the CSB constructor for that which is required.

Also, I like how you did this but just to point out that there's a method URIBuilder.fromUri() and then you could just set the truncated host.

**Review by yihezkel — CHANGES_REQUESTED** [review]

File: data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java (hunk 2 above)
Comment (ohadbitt) [reply data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java:165]: what could be a negative test here?

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java (hunk 1 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java:45]: we should consider this but yea not for this PR

File: data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java (hunk 2 above)
Comment (yihezkel) [reply data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java]: void dontChangeUrlIfNotFederated() throws URISyntaxException {
        String clusterUrl = "https://service.uri";
        ConnectionStringBuilder csb = ConnectionStringBuilder.createWithAadApplicationCredentials(clusterUrl, "id", "appKey");
        ClientImpl client = new ClientImpl(csb);
        Assertions.assertEquals(clusterUrl, csb.getClusterUrl());
    }

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 4]
```diff
@@ -33,7 +35,16 @@
     private final String applicationNameForTracing;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
-        clusterUrl = csb.getClusterUrl();
+        String url = csb.getClusterUrl();
+        URI clusterUri = new URI(url);
+        String host = clusterUri.getHost();
+        String auth = clusterUri.getAuthority().toLowerCase();
```
[the remark this answers is not in the document]
Comment (yihezkel) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java:41]: My mistake, I thought we could create a CSB using a constructor and bypass the helper method (createWithAadApplicationCredentials()). IIRC that's valid in the .NET SDK.
I see here that the constructor is private.

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 3 above)
Comment (yihezkel) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: You're right, CSB can only be created via helper method, as above.

**Review by yihezkel — APPROVED** [review]
